jwt/jwt.py

```diff
@@ -48,12 +51,23 @@ def encode(self, payload: dict, key: AbstractJWKBase = None, alg='HS256',
 
     def decode(self, message: str, key: AbstractJWKBase = None,
                do_verify=True, algorithms: AbstractSet[str]=None) -> dict:
+        # utc now with timezone
+        now = get_time(get_time(datetime.utcnow()))
         try:
             message_bin = self._jws.decode(message, key, do_verify, algorithms)
         except JWSDecodeError as why:
             raise JWTDecodeError('failed to decode JWT') from why
         try:
             payload = json.loads(message_bin.decode('utf-8'))
+            if 'exp' in payload:
+                exp = get_time(payload.get('exp'))
+                if now > exp:
+                    raise JWTDecodeError("JWT Expired")
```

Comment:
I think it is need to support a use case to decode expired or not-valid-yet JWTs. So can you implement a switch like `do_time_check=False` (If we choose `True` as the default value, it breaks backward compatibility.) ?

Reply:
And also `exp` and `nbf` in the `payload` are not guaranteed to be `int`. It is not an int, `get_time` would not return `datetime` and now < nbf raises a `TypeError`.